README.md

```diff
@@ -46,23 +79,66 @@ const resolved = await resolver.resolve({
   }
 });
 
-console.log(resolved.result);
-
-// ==> outputs the original object, with local refs resolved and replaced
-//
-// {
-//   user: {
-//     name: 'json'
-//   },
-//   models: {
-//     user: {
-//       name: 'john'
-//     }
-//   }
-// }
+// ==> result is the original object, with local refs resolved and replaced
+expect(resolved.result).toEqual({
+  user: {
+    name: "json"
+  },
+  models: {
+    user: {
+      name: "john"
+    }
+  }
+});
 ```
 
-#### With Authority Readers
+#### Example: Resolve a Subset of the Source
+
+This will resolve the minimal number of references needed for the given target, and return the target.
+
+In the example below, the address reference (`https://slow-website.com/definitions#/address`) will NOT be resolved, since
+it is not needed to resolve the `#/user` jsonPointer target we have specified. However, `#/models/user/card` IS resolved since
+it is needed in order to full resolve the `#/user` property.
+
+```ts
+import { Resolver } from "@stoplight/json-ref-resolver";
+
+const resolver = new Resolver();
+const resolved = await resolver.resolve(
+  {
+    user: {
+      $ref: "#/models/user"
+    },
+    address: {
+      $ref: "https://slow-website.com/definitions#/address"
```

Comment:
Let's say this reference needed to be resolved. What would the http request look like? What URL would have been hit? `https://slow-website.com/definitions.json` and then resolving `#/address`?

Reply:
Yup!